src/main/java/org/spongepowered/api/entity/living/player/Player.java

```diff
@@ -339,4 +340,31 @@ default GameModeData getGameModeData() {
      */
     void setSpectatorTarget(@Nullable Entity entity);
 
+    /**
+     * Gets the {@link WorldBorder} for this player. If no border is set, a
```

Comment:
I think it would be better to return an `Optional<WorldBorder>` in this method that will only return if there is a custom player world border set. Otherwise this could easily lead to confusion like in your test plugin: If you modify the default world border returned by this method you actually modify the main world border.